p2p/discovery/refresher.go

```diff
@@ -62,28 +61,33 @@ func (r *refresher) Bootstrap(ctx context.Context, minPeers int) error {
 	// if there were any results but we didn't reach minPeers it will try the same procedure
 	// using the results as servers to query. every failed loop will wait a backoff
 	// to let other nodes populate before flooding with queries.
-
+	seed := false
+	size := r.book.NumAddresses()
+	if size == 0 {
+		r.book.AddAddresses(r.bootNodes, r.localAddress)
+		servers = r.bootNodes
+		seed = true
+	} else {
+		srv := r.book.AddressCache()
+		servers = srv[:common.Min(minPeers, len(srv))]
```

Comment:
This strikes a bit odd - what does the number of servers have to do with `minPeers`?

Reply:
`requestAddresses` won't stop until it will request peers from all server it got. it will do so 3 servers at a time but still, we don't want to pass it 1000 peers if we have. so we can use the requested amount of peers as servers. or if we only have less than that amount we'll use all of them.